### DISCUSSION TITLE
How to handle Ctrl+C/SIGINT with own code

### DISCUSSION THREAD

**Discussion by Whanos** [rwf2/Rocket#2395] [Questions] [answered]
Hey there,

I tried using the ctrlc crate to handle SIGINT so I can add my own code to it (to save some values), but it seems that rocket's handler is overriding it.

What's the recommended way to add my own code to Rocket's shutdown sequence so I can save these values?

Thanks

**Comment by SergioBenitez** [comment, marked as answer]
You simply need to disable Rocket's default handler. See the config guide and API docs for how: https://rocket.rs/v0.5-rc/guide/configuration/ and https://api.rocket.rs/v0.5-rc/rocket/config/struct.Shutdown.html#example. If you want to shutdown as well, then you'll need to call shutdown() from your signal handler.

**Reply by Whanos** [reply]
Hi Sergio, I figured it out from this.

Thank you for your help!
